tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -352,6 +352,9 @@ public:
    NTupleSize_t GetNEntries() const;
    NTupleSize_t GetNElements(DescriptorId_t columnId) const;
 
+   /// Returns the root field of this NTuple. The root field is the parent of all top-level
+   /// data fields.
+   DescriptorId_t GetRootFieldId() const;
```

Comment:
Prefixed this method with `Get` instead of `Find` since this is supposed to be infallible, NTuples should always have a root field. In contrast, find methods could return `ROOT::Experimental::kInvalidDescriptorId`.

Reply:
Given the name of the project (ROOT :) ), maybe we ought to find an alternative name?

Reply:
Good point, ROOT is already pretty overloaded :)
Maybe origin or base? 
* `GetOriginFieldId`
* `GetBaseFieldId`

Reply:
Base might be confused with base class.

Is this the 'parent' field or the highest/externalest field?

If it is the later then maybe ```TopLevel``` (or indeed ```Origin```).  @jblomer what do you think?

Reply:
I think `TopLevel` fits well, or perhaps simply `GetTopFieldId()`

Reply:
The "root field" naming is also used in `RNTupleModel`: 
https://github.com/root-project/root/blob/d0ed0d000d787176d13f6907f10d43c9e024be94/tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx#L91
and as a typename: 
https://github.com/root-project/root/blob/d0ed0d000d787176d13f6907f10d43c9e024be94/tree/ntuple/v7/inc/ROOT/RField.hxx#L287-L288

Reply:
In my opinion, we ought to change them too :)

Reply:
Yes, I think that makes sense. A small issue with `TopLevel` is that we've been using that to refer to the actual `ntuple` data fields, but we can of course use different names for those too. 

```text
{ // current "root field"
  // top-level fields
  "floats": { ... },      
  "bool_vec": { ... }
}
```

Reply:
Philippe suggested to make a separate PR for the name changes away from `Root...`. I think that's a good idea. As for this comment, how about `top-level field` (former root field) and `first-level fields`?

Reply:
You got it, new PR is here: #5805

Reply:
> first-level fields

In a similar context we have use "terminal branch/leaf".

Reply:
Just discussed again with Max.  How about this: let's rename `root field` to `head field` and call the actual data root fields `top-level fields`.

Reply:
Sorry, got lost.  What is the difference between "head field" and "top-level fields"?

Reply:
The head field is the "highest, most external" field (what used to be called the root field, like a tree's root node). Every NTuple is guaranteed to have one (and only one), but it has no associated data (similar to dummy leaf nodes in a tree that carry no information but make insertions easier). Whenever we traverse the NTuple, we start at the head field and visit its children. 

The direct children of the head field are what we're calling top-level fields. These hold the actual data of the NTuple. I think it is good to have a term for these, because this is what an NTuple really is made of (no one really cares about the head field, there's nothing there). Top-level fields can have child fields, etc., but top-level fields get a name because they're the "level" we do any useful work at. 

```text
     (head field)
    /     |     \
   /      |      \
"ints"  "bools" "structs"   <- the "top-level" fields
[0]     [true]   [MyClass a]
[100]   [true]   [MyClass b]
[401]   [false]  [MyClass c]
```

Reply:
thanks for the clarification.

Reply:
You are very welcome :). The collision between "head" and "top-level" is unfortunate. I am happy to iterate on #5805 until we find a name that everyone likes. My preference is to keep "top-level" fields the same and change up "head field". 

Is reverting to `RootField / FieldRoot` out of the question? Perhaps in this context it is OK, because of the established meaning of root node for trees.

We could also call it `FieldRoot`, which to me seems more explicit (the root of all fields). 
The only method would called `GetFieldRootId()`, implying a unique value is returned.